terraform-tests/postgres_test.go

```diff
@@ -55,227 +59,218 @@ var _ = Describe("postgres", Label("postgres-terraform"), func() {
 		"aws_secret_access_key":           awsSecretAccessKey,
 	}
 
-	Describe("postgres provision", func() {
-		var terraformProvisionDir string
-		var plan tfjson.Plan
-		BeforeEach(OncePerOrdered, func() {
-			terraformProvisionDir = path.Join(workingDir, "postgresql/provision")
-			Init(terraformProvisionDir)
-		})
-
-		Context("provisions an instance", Ordered, func() {
-			Context("postgres parameter groups", func() {
-
-				Context("No parameter group name passed", func() {
-					BeforeEach(func() {
-						plan = ShowPlan(terraformProvisionDir, buildVars(defaultVars, map[string]any{
-							"postgres_version": "14.1",
-						}))
-					})
-
-					It("should create a parameter group", func() {
-						Expect(ResourceCreationForType(plan, "aws_db_parameter_group")).To(HaveLen(1))
-						Expect(AfterValuesForType(plan, "aws_db_instance")).To(
-							MatchKeys(IgnoreExtras, Keys{
-								"parameter_group_name": Equal("rds-pg-csb-postgresql-test"),
-							}))
-						Expect(AfterValuesForType(plan, "aws_db_parameter_group")).To(
-							MatchKeys(IgnoreExtras, Keys{
-								"name":   Equal("rds-pg-csb-postgresql-test"),
-								"family": Equal("postgres14"),
-								"parameter": ConsistOf(MatchKeys(IgnoreExtras, Keys{
-									"name":  Equal("rds.force_ssl"),
-									"value": Equal("0"),
-								}))}))
-					})
-					Context("When requiring SSL", func() {
-						BeforeEach(func() {
-							plan = ShowPlan(terraformProvisionDir, buildVars(defaultVars, map[string]any{
-								"require_ssl": true,
-							}))
-						})
-
-						It("should configure the parameter in the parameter group", func() {
-							Expect(AfterValuesForType(plan, "aws_db_parameter_group")).To(
-								MatchKeys(IgnoreExtras, Keys{
-									"parameter": ConsistOf(MatchKeys(IgnoreExtras, Keys{
-										"name":  Equal("rds.force_ssl"),
-										"value": Equal("1"),
-									}))}))
-						})
-
-					})
-
-				})
+	BeforeAll(func() {
+		terraformProvisionDir = path.Join(workingDir, "postgresql/provision")
+		Init(terraformProvisionDir)
+	})
 
-				Context("Parameter group passed", func() {
+	Context("postgres parameter groups", func() {
+		When("no parameter group name passed", func() {
+			BeforeAll(func() {
```

Comment:
**nitpick**: It's strange to see `BeforeAll` followed by a single test case. Wouldn't it be simpler to move the `ShowPlan` to a `JustBeforeEach` at the top, and then set the appropriate variables in respective sub-sections? This should also result in a clearer setup intent.

Reply:
Yes, that's a good idea. I'll look to do that in a followup PR.